tests/e2e/performance/test_multi_snapshots.py

```diff
@@ -0,0 +1,424 @@
+log_names = {'start': None, 'end': []}
+
+snap_yaml = None  # define this parameter as global
+format = '%H:%M:%S.%f'
+
+log_file_name = os.path.basename(__file__).replace('.py', '.log')
+
+
+def msg_logging(msg):
+    """
+    This function is logging the message to the log file, and also print it
+    for the caller script output
+
+    Args:
+        msg (str): The message to log as info and print on the console
+
+    """
+    print(msg)
+    log.info(msg)
+
+
+def run_command(cmd):
+    """
+    Running command on the OS and return the STDOUT & STDERR outputs
+    in case of argument is not string or list, return error message
+
+    Args:
+        cmd (str/list): the command to execute
+
+    Returns:
+        list : all STDOUT / STDERR output as list of lines
+
+    """
+    if isinstance(cmd, str):
+        command = cmd.split()
+    elif isinstance(cmd, list):
+        command = cmd
+    else:
+        return ERRMSG
+
+    msg_logging(f'Going to run {cmd}')
+    cp = subprocess.run(
+        command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        stdin=subprocess.PIPE,
+        timeout=600
+    )
+    output = cp.stdout.decode()
+    err = cp.stderr.decode()
+    # exit code is not zero
+    if cp.returncode:
+        log.error(f'Command finished with non zero ({cp.returncode}) {err}')
+        print(f'Command finished with non zero ({cp.returncode}) {err}')
+        output += f'{ERRMSG} {err}'
+
+    output = output.split('\n')  # convert output to list
+    output.pop()  # remove last empty element from the list
+    return output
+
+
+def run_oc_command(cmd, namespace):
+    """
+    Running an 'oc' command
+
+    Args:
+        cmd (str): the command to run
+        namespace (str): the namespace where to run the command
+
+    Returns:
+        list : the results of the command as list of lines
+
+    """
+    command = (f'oc --kubeconfig {params["KUBECONFIG"]} -n {namespace} {cmd}')
+    return run_command(command)
+
+
+def run_cmd_on_pod(command, pod_name):
+    """
+    Execute a command on a pod (e.g. oc rsh)
+
+    Args:
+        command (str): The command to execute on the given pod
+        pod_name (str): the pod name to execute the command on
+
+    Returns:
+        list: the command output as list of lines
+
+    """
+    rsh_cmd = f'rsh {pod_name} {command}'
+    return run_oc_command(rsh_cmd, params['NSPACE'])
+
+
+def get_env_args():
+    """
+    Checking that all arguments need for this script, defined as environment
+    variables
+
+    """
+    print(f'Validating arguments : {params.keys()}')
+    error = 0
+    for key in params.keys():
+        params[key] = os.getenv(key)
+        if params[key] is None:
+            error = 1
+            print(f'Error: {key} is not define !')
+        else:
+            print(f'{key} - {params[key]}')
+    if error:
+        print('Not all variables defined !')
+        sys.exit(error)
+
+    full_log = params['LOGPATH'] + '/' + log_file_name
```

Comment:
Use f-string for string concatenation instead of + operator.
Follow his in rest of this file, wherever applicable.

Reply:
changed